src/analysis/calculus/cont_diff.lean

```diff
@@ -2264,3 +2265,308 @@ lemma cont_diff.restrict_scalars (h : cont_diff 𝕜' n f) :
 cont_diff_iff_cont_diff_at.2 $ λ x, h.cont_diff_at.restrict_scalars _
 
 end restrict_scalars
+
+/-!## Quantitative bounds -/
+
+section
+
+open finset
+
+/- TODO porting note: move to the file `data.nat.choose.sum` -/
+/-- The sum of `(n+1).choose i * a i * b (n+1-i)` can be split into two sums at rank `n`,
+respectively of `n.choose i * a i * b (n+1-i)` and `n.choose i * a (i+1) * b (n-i)`. -/
+lemma sum_choose_succ_mul_mul_sub {R : Type*} [comm_semiring R] (a b : ℕ → R) (n : ℕ) :
+  ∑ i in range (n+2), ((n+1).choose i : R) * a i * b (n + 1 - i) =
+    ∑ i in range (n+1), (n.choose i : R) * a i * b (n + 1 - i)
+      + ∑ i in range (n+1), (n.choose i : R) * a (i + 1) * b (n - i) :=
```

Comment:
The antidiagonal version of this is
```lean
lemma sum_antidiagonal_choose_succ {R : Type*} [comm_semiring R] (a b : ℕ → R) (n : ℕ) :
  ∑ ij in nat.antidiagonal n.succ,
    ↑((n + 1).choose ij.1) * a ij.1 * b ij.2 =
    ∑ ij in nat.antidiagonal n, ↑(n.choose ij.1) * a ij.1 * b (ij.2 + 1)
      + ∑ ij in nat.antidiagonal n, ↑(n.choose ij.1) * a (ij.1 + 1) * b ij.2 :=
```
which is nicely symmetric, although I don't know whether it's easier to prove.

Reply:
I have added the antidiagonal version. Definitely not easier to prove (so I've deduced it from the other one) but nicely symmetric, as you say!